File: bindings/go/binding_test.go
package tree_sitter_gotmpl_test

import (
	"testing"

	tree_sitter_gotmpl "github.com/infastin/tree-sitter-gotmpl/bindings/go"
	tree_sitter "github.com/tree-sitter/go-tree-sitter"
)

func TestCanLoadGoTmplGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_gotmpl.LanguageGoTmpl())
	if language == nil {
		t.Errorf("Error loading Go Template grammar")
	}
}

func TestCanLoadGoYAMLTmplGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_gotmpl.LanguageGoYAMLTemplate())
	if language == nil {
		t.Errorf("Error loading Go YAML Template grammar")
	}
}

File: bindings/node/binding_test.js
const assert = require('node:assert');
const { test } = require('node:test');

const Parser = require('tree-sitter');

test('can load Go template grammar', () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(require('./gotmpl')));
});

test('can load Helm grammar', () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(require('./helm')));
});
